avm/res/sql/server/main.bicep

```diff
@@ -249,36 +250,53 @@ module server_databases 'database/main.bicep' = [
-      zoneRedundant: database.?zoneRedundant ?? true
-      elasticPoolId: database.?elasticPoolId ?? ''
-      backupShortTermRetentionPolicy: database.?backupShortTermRetentionPolicy ?? {}
-      backupLongTermRetentionPolicy: database.?backupLongTermRetentionPolicy ?? {}
-      createMode: database.?createMode ?? 'Default'
-      sourceDatabaseResourceId: database.?sourceDatabaseResourceId ?? ''
-      sourceDatabaseDeletionDate: database.?sourceDatabaseDeletionDate ?? ''
-      recoveryServicesRecoveryPointResourceId: database.?recoveryServicesRecoveryPointResourceId ?? ''
-      restorePointInTime: database.?restorePointInTime ?? ''
+
+      // properties from the databse object. If not provided, defaults specified in the child resource will be used
+      name: database.name
+      sku: database.?sku
+      autoPauseDelay: database.?autoPauseDelay
+      availabilityZone: database.?availabilityZone
+      catalogCollation: database.?catalogCollation
+      collation: database.?collation
+      createMode: database.?createMode
+      elasticPoolResourceId: database.?elasticPoolResourceId
+      encryptionProtector: database.?encryptionProtector
+      encryptionProtectorAutoRotation: database.?encryptionProtectorAutoRotation
+      federatedClientId: database.?federatedClientId
+      freeLimitExhaustionBehavior: database.?freeLimitExhaustionBehavior
+      highAvailabilityReplicaCount: database.?highAvailabilityReplicaCount
+      isLedgerOn: database.?isLedgerOn
+      licenseType: database.?licenseType
+      longTermRetentionBackupResourceId: database.?longTermRetentionBackupResourceId
+      maintenanceConfigurationId: database.?maintenanceConfigurationId
+      manualCutover: database.?manualCutover
+      maxSizeBytes: database.?maxSizeBytes
+      minCapacity: database.?minCapacity
+      performCutover: database.?performCutover
+      preferredEnclaveType: database.?preferredEnclaveType
+      readScale: database.?readScale
+      recoverableDatabaseResourceId: database.?recoverableDatabaseResourceId
+      recoveryServicesRecoveryPointResourceId: database.?recoveryServicesRecoveryPointResourceId
+      requestedBackupStorageRedundancy: database.?requestedBackupStorageRedundancy
+      restorableDroppedDatabaseResourceId: database.?restorableDroppedDatabaseResourceId
+      restorePointInTime: database.?restorePointInTime
+      secondaryType: database.?secondaryType
+      sourceDatabaseDeletionDate: database.?sourceDatabaseDeletionDate
+      sourceDatabaseResourceId: database.?sourceDatabaseResourceId
+      sourceResourceId: database.?sourceResourceId
+      useFreeLimit: database.?useFreeLimit
+      zoneRedundant: database.?zoneRedundant
+
+      // diagnosticSettings: database.?diagnosticSettings
+      // sampleName: database.?sampleName ?? ''
+      // backupShortTermRetentionPolicy: database.?backupShortTermRetentionPolicy ?? {}
+      // backupLongTermRetentionPolicy: database.?backupLongTermRetentionPolicy ?? {}
```

Comment:
Leftover

Reply:
Actually I forgot to uncomment them after defining the objects. Good catch.